src/components/Cluster/ClusterDetail/NodePool.tsx

```diff
@@ -165,39 +234,20 @@ class NodePool extends Component {
   }
 }
 
-NodePool.propTypes = {
-  availableZonesGridTemplateAreas: PropTypes.string,
-  cluster: PropTypes.object,
-  nodePool: PropTypes.shape({
-    availability_zones: PropTypes.any, // TODO fix it.
-    id: PropTypes.string,
-    name: PropTypes.string,
-    node_spec: PropTypes.object,
-    scaling: PropTypes.shape({
-      min: PropTypes.number,
-      max: PropTypes.number,
-    }),
-    status: PropTypes.shape({
-      nodes: PropTypes.number,
-      nodes_ready: PropTypes.number,
-    }),
-  }),
-  nodePoolActions: PropTypes.object,
-  dispatch: PropTypes.func,
-  provider: PropTypes.string,
-};
-
-function mapStateToProps(state, ownProps) {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function mapStateToProps(state: Record<string, any>, ownProps: INodePoolProps) {
```

Comment:
```suggestion
function mapStateToProps(state: Record<string, any>, ownProps: INodePoolProps): IStateProps {
```

Reply:
`IStateProps` have more properties defined...

Does it makes sense to have a type for this if we are defining this prop in the props interface?

Reply:
But `IStateProps` is supposed to contain only the props that are retrieved from the redux state.
It is useful to have them defined separately for each component (`INodePoolProps`, `IStateProps`, `IDispatchProps`), because it's easier to know while developing where the props are coming from (if it's an own prop, or an action, or a prop coming from the state)

Reply:
👍 ok, I get it, thanks